### PR TITLE
LibGfx + LibGUI + LibWeb: Add support for underlining using squiggly lines

### PR DESCRIPTION
**Pull request by TobyAsE**

### CODE REVIEW

File: Userland/Libraries/LibGfx/Painter.cpp [hunk 1]
```diff
@@ -1825,6 +1826,25 @@ void Painter::draw_line(IntPoint const& a_p1, IntPoint const& a_p2, Color color,
     }
 }
 
+void Painter::draw_triangle_wave(const IntPoint& a_p1, const IntPoint& a_p2, Color color, int amplitude, int thickness)
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibGfx/Painter.cpp:1829]: Nitpick: west const!

File: Userland/Libraries/LibGfx/Painter.h [hunk 2]
```diff
@@ -59,6 +59,7 @@ class Painter {
     void set_pixel(IntPoint const&, Color);
     void set_pixel(int x, int y, Color color) { set_pixel({ x, y }, color); }
     void draw_line(IntPoint const&, IntPoint const&, Color, int thickness = 1, LineStyle style = LineStyle::Solid, Color alternate_color = Color::Transparent);
+    void draw_triangle_wave(const IntPoint&, const IntPoint&, Color color, int amplitude, int thickness = 1);
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibGfx/Painter.h:62]: Nitpick: west const!

**Review by AtkinsSJ — CHANGES_REQUESTED** [review]
A couple of nit picks, but this is very cool! :grinning:

File: Userland/Libraries/LibGfx/Painter.cpp (hunk 1 above)
Comment (TobyAsE) [reply Userland/Libraries/LibGfx/Painter.cpp:1829]: I got bitten by the copy paste bugs!

File: Userland/Libraries/LibGfx/TextAttributes.h [hunk 3]
```diff
@@ -12,10 +13,18 @@
...
 
 struct TextAttributes {
+    enum class UnderlineStyle {
+        Solid,
+        Wavy
+    };
+
     Color color;
-    Optional<Color> background_color;
+    Optional<Color> background_color {};
```
Comment (alimpfard) [inline Userland/Libraries/LibGfx/TextAttributes.h:22]: The braces are not needed here.

File: Userland/Libraries/LibGfx/TextAttributes.h [hunk 4]
```diff
@@ -12,10 +13,18 @@
...
     bool underline { false };
     bool bold { false };
+
+    Optional<Color> underline_color {};
```
Comment (alimpfard) [inline Userland/Libraries/LibGfx/TextAttributes.h:26]: Nor here.

**Review by alimpfard — COMMENTED** [review]
This is a nice first implementation, but for it to be nice to use, we'll have to expand on it a bit (as we discussed on discord), though I don't see any reason to require those as part of this PR.

So all that remains imo, is just a few nitpicks, plus whatever the CI is sad about (which looks like a flake)

File: Userland/Libraries/LibGfx/TextAttributes.h (hunk 3 above)
Comment (TobyAsE) [reply Userland/Libraries/LibGfx/TextAttributes.h:22]: There are some places that create `TextAttributes` without specifying (or need to) those Optionals. So I just empty-initialize them here to not do that at every single use site.
The only thing that is really needed is `color` anyways

Comment (alimpfard) [reply Userland/Libraries/LibGfx/TextAttributes.h:22]: `Optional<T>`'s default ctor initialises the...ah, I see what you mean, yeah sure.

**Pull request merged**